docs/pages/guides/localization.mdx

```diff
@@ -0,0 +1,385 @@
+### Full Demo
+
+<SnackInline label="Localization" dependencies={['expo-localization', 'i18n-js']}>
+
+```tsx
+import { View, StyleSheet, Text } from 'react-native';
+import * as Localization from 'expo-localization';
+import { I18n } from 'i18n-js';
+
+// Set the key-value pairs for the different languages you want to support.
+const translations = {
+  en: { welcome: 'Hello', name: 'Charlie' },
+  ja: { welcome: 'こんにちは' },
+};
+const i18n = new I18n(translations);
+
+// Set the locale once at the beginning of your app.
+i18n.locale = Localization.locale;
+
+// When a value is missing from a language it'll fall back to another language with the key present.
+i18n.enableFallback = true;
+// To see the fallback mechanism uncomment the line below to force the app to use the Japanese language.
+// i18n.locale = 'ja';
+
+export default function App() {
+  return (
+    <View style={styles.container}>
+      <Text style={styles.text}>
+        {i18n.t('welcome')} {i18n.t('name')}
+      </Text>
+      <Text>Current locale: {i18n.locale}</Text>
+      <Text>Device locale: {Localization.locale}</Text>
+    </View>
+  );
+}
+
+/* @hide const styles = StyleSheet.create({ ... }); */
+const styles = StyleSheet.create({
+  container: {
+    backgroundColor: '#fff',
+    alignItems: 'center',
+    justifyContent: 'center',
+    flex: 1,
+  },
+  text: {
+    fontSize: 20,
+    marginBottom: 16,
+  },
+});
+/* @end */
+```
+
+</SnackInline>
+
+
+<Collapsible summary="Using other translation libraries">
+
+
+
+While this guide uses `i18n-js` as an example, there are many other libraries that can help you with translations, and they each have different features:
+
+- Creating translations is a large effort that you'd ideally want to hire someone for, but it's hard for them to edit translations directly in your code. Some translation libraries can integrate with translation management tools (essentially a web service that lets you outsource, auto-generate or just makes it easier to create translations).
+
+- Some libraries make it easy to move the component structure around (and not just words of your copy) based on translation strings. Imagine you want to localize a string that includes a `<Pressable>` link, and you want the link to be in a different order from the rest of the text depending on the translation.
+
+Here are some libraries that can be a good option depending on your needs:
+  
+- [`React i18next`](https://react.i18next.com/)
+
+  `React i18next` is a stable, well-maintained library based on `i18next`.
+
+- [`typesafe-i18n`](https://github.com/ivanhofer/typesafe-i18n)
+
+  `typesafe-i18n` compiles translations so that it saves space in the production bundle. It also generates typescript types for your translations, so that you can use them in your code and get autocomplete and type safety.
```

Comment:
```suggestion
  `typesafe-i18n` compiles translations to save space in the production bundle. It also generates typescript types for your translations so that you can use them in your code and get autocomplete and type safety.
```